vpr/src/analytical_place/global_placer.h

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @author  Alex Singer
+ * @date    October 2024
+ * @brief   The declarations of the Global Placer base class which is used to
+ *          define the functionality of all global placers in the AP flow.
+ *
+ * Global Placers create a Partial Placement given onlythe netlist and the
+ * architecture. It uses analytical techniques to find a placement that
+ * optimizes for objectives subject to some of the constraints of the FPGA
+ * architecture.
+ */
+
+#pragma once
+
+#include <memory>
+
+// Forward declarations
+class APNetlist;
+class AnalyticalSolver;
+class PartialPlacement;
+class PartialLegalizer;
+
+/**
+ * @brief Enumeration of all of the global placers currently implemented in VPR.
+ */
+enum class e_global_placer {
+    SimPL       // Global placer based on the SimPL paper.
+};
+
+/**
+ * @brief The Global Placer base class
+ *
+ * This declares the functionality that all Global Placers will use. This
+ * provides a standard interface for the global placers so they can be used
+ * interchangably. This makes it very easy to test and compare different global
+ * placers.
+ */
+class GlobalPlacer {
+public:
+    virtual ~GlobalPlacer() {}
+
+    /**
+     * @brief Constructor of the base GlobalPlacer class
```

Comment:
Might want to briefly say what the parameters are. For example:
@param netlist:  netlist of the design at some abstraction level; typically this would have atoms and groups of atoms locked together by carry chains etc. as its blocks.